File: 1603-running-sum-of-1d-array/running-sum-of-1d-array.cpp
class Solution {
public:
    vector<int> runningSum(vector<int>& nums) {
        if (nums.size() == 1){
            return nums;
        }
        vector<int> cum_sum;
        cum_sum.push_back(nums[0]);
        for (int i = 1; i < nums.size(); i++){
            cum_sum.push_back(cum_sum.back() + nums[i]);
        }    
        return cum_sum;    
    }
};

File: 1603-running-sum-of-1d-array/running-sum-of-1d-array.py
class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:
        run_sum = [None for i in range(len(nums))]
        run_sum[0] = nums[0]
        for i in range(1, len(nums)):
            run_sum[i] = run_sum[i-1] + nums[i]
        return list(run_sum)